yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -169,6 +172,24 @@ private[yarn] class YarnAllocator(
   }
 
   /**
+   * Init `executorIdCounter`
+   */
+  def initExecutorIdCounter(): Unit = {
```

Comment:
actually, you can just initialize `executorIdCounter` to the value you get from the driver + 1 right?
```
/**
 * [Add huge comment related to SPARK-12864 to explain why we need to do this here.]
 */
private var executorCounter: Int = {
  driverRef.askWithRetry[Int](RetrieveMaxExecutorId) + 1
}
```